content/docs/getting-started/futures.md

```diff
@@ -1,19 +1,72 @@
 ---
 title: "Futures"
-weight : 1030
+weight : 1020
 menu:
   docs:
     parent: getting_started
 ---
 
-Futures, hinted at earlier in the guide, are the building block used to manage
-asynchronous logic. They are the underlying asynchronous abstraction used by
-Tokio.
+Let's take a closer look at futures. Tokio is built on top of the [`futures`] crate
+and uses its runtime model. This allows it to interop with other libraries also using
+the [`futures`] crate.
 
-The future implementation is provided by the [`futures`] crate. However, for
-convenience, Tokio re-exports a number of the types.
+**Note**: This runtime model is very different than async libraries found in
+other languages. While, at a high level, APIs can look similar, the way code
+gets executed differs.
 
-# What Are Futures?
+We'll be taking a closer look at the runtime in the next section, but a basic
+understanding of the runtime is necessary to understand futures. To gain this
+understanding, we'll firs tlook at the synchronous model that Rust uses by default
+and see how this differs from Tokio's asynchronous model.
+
+# Synchronous Model
+
+First, let's talk briefly about the synchronous (or blocking) model. This is the
+model that the Rust [standard library] uses.
+
+```rust
+# use std::io::prelude::*;
+# use std::net::TcpStream;
+# fn dox(mut socket: TcpStream) {
+// let socket = ...;
+let mut buf = [0; 1024];
+let n = socket.read(&mut buf).unwrap();
+
+// Do something with &buf[..n];
+# }
+```
+
+When `socket.read` is called, either the socket has pending data in its receive
+buffer or it does not. If there is pending data, then the call to `read` will
+return immediately and `buf` will be filled with that data. However, if there is
+no pending data, then the `read` function will block the current thread until
+data is received. At which time, `buf` will be filled with this newly received
+data and the `read` function will return.
+
+In order to perform reads on many different sockets concurrently, a thread per
+socket is required. Using a thread per socket does not scale up very well to
+large numbers of sockets. This is known as the [c10k] problem.
+
+# Non-blocking sockets
+
+The way to avoid blocking a thread when performing an operation like read is to
```

Comment:
"The way to avoid blocking a thread when performing an operation like read is to not block the thread!"

Probably can be phrased better!

"Non-blocking sockets allow performing operations, like read, without blocking the thread".